index.js

```diff
@@ -99,13 +103,37 @@ var pcbStackup = function(layers, options, done) {
       layerOptions.plotAsOutline = options.outlineGapFill
     }
 
-    var converter = gerberToSvg(layer.gerber, layerOptions, finishLayer)
+    var converterDone = function() {
+      layer.done = true
+    }
+
+    var converter
+
+    if (layer.converter != null && layer.gerber == null) {
+      converter = layer.converter
+      converterDone()
+    }
+    else {
+      converter = gerberToSvg(layer.gerber, layerOptions, converterDone)
+    }
 
     stackupLayers.push({
       type: layerType,
       converter: converter,
       options: layerOptions
     })
+
+    var finishIfDone = function() {
```

Comment:
I don't think it's necessary to make our own polling loop; javascript does that for us. We've already got the `finishLayer` callback, let's just use it (and eliminate the need for a done flag)

``` js
var converter = layer.converter || gerberToSvg(layer.gerber, layerOptions, finishLayer)

stackupLayers.push({
  type: layerType,
  converter: converter,
  options: layerOptions
})

if (layer.converter) {
  finishLayer()
}
```